cmd/gossamer/config.go

```diff
@@ -465,6 +434,33 @@ func setDotGlobalConfigFromToml(tomlCfg *ctoml.Config, cfg *dot.GlobalConfig) {
 	}
 }
 
+func setDotBasepathConfig(ctx *cli.Context, tomlConfig *ctoml.Config, cfg *dot.GlobalConfig) {
+	setDotBasepathConfigFromToml(tomlConfig, cfg)
+	setDotBasepathConfigFromFlags(ctx, cfg)
+}
```

Comment:
nit it would be nice to have these few functions as methods on `dot.GlobalConfig` instead.

Alternatively you should change those functions to pass in values and return values, that makes it a bit easier to read.

i.e.

```go
func setDotBasepathConfig(ctx *cli.Context, tomlConfig *ctoml.Config, cfg dot.GlobalConfig) (updatedCfg dot.GlobalConfig)
```

Reply:
I think I'm going to omit that change from this PR, but I'll keep it in mind as I continue to improve the CLI stuff.